CHANGELOG.md

```diff
@@ -1,5 +1,94 @@
+* Users can now add arbitrary files to the installer using the ``extra_files`` keyword. (#465 via #500)
+* Added two new keys, `initialize_conda` and `register_python`, to control whether these options
+  should be offered in the installer or not. (#507)
+* Add support for multi-environment installs via ``extra_envs`` keyword (#359 via #509, #553, #599)
+* Enable alternative solvers by obeying `CONDA_SOLVER` if set and available. (#531, #597)
+* PKG installers now default to the macOS logging system to log messages.
+  UI notifications can be enabled with `progress_notifications`
+  (off by default). (#535)
+* The graphical macOS installer now also displays the version number of the software in the window title. (#536)
+* Enable ``conclusion_text`` on Windows ``.exe`` and Unix ``.sh`` installers (#443 via #547 and #550).
+* All installers support pre- and post-install scripts and expose the same environment variables:
+  `PREFIX`, `INSTALLER_NAME`, `INSTALLER_VER`, `INSTALLER_PLAT`, `INSTALLER_TYPE`.
+  The `pre_install_desc` key is now available, fulfilling the same role as `post_install_desc` (#556 via #558)
+* Bypass the solver by using an `@EXPLICIT` input file for `conda install` commands. (#541 via #559)
+* cache files have correct creation and modification timestamps for
+  mamba which looks at the timestamp instead of the _mod value in the
+  cache json (#579).
+* Windows GUI installer enhancement to allow for additional custom pages. These new pages can be added after the welcome page and before the conclusion. These extra pages can display text, links, and images. Such pages can be useful in providing additional instructions, guidance, or promotional materials for end-users before they begin using the application they just installed. (#590)
+* A new key `build_outputs` allows to generate extra artifacts besides the installer,
+  like JSON metadata files, solved environments lock files, or licensing reports (#595, #602).
+* Improve and publish the documentation to `conda.github.io`. (#437, #598)
+* header.sh and osx scripts hardening by adding `set -eu` (sh) / `set -euo pipefail` (bash) and fixing all shellcheck findings. Shell scripts don't stop per default when commands finish with an error causing masked errors and undefined behaviours. `set -e` changes that behaviour by stoping in all cases where errors happen enabling better error reports on the actual error. If an error should be ignored then a command can be run via `$cmd || true`. Two test cases running shellcheck ensure that future changes get tested.
+
+### Bug fixes
+
+* Check `makensis.exe` exit code in verbose mode too (#453 via #475)
+* Fix constructor failures when the repo/subchannel only has ``noarch`` packages. (#512)
+* Fix logging error that would make ``constructor`` crash if ``verbose`` mode was enabled. (#534)
+* Hardcode paths to Apple tools (``productbuild``, ``productsign``, ``codesign``) to avoid using other tools with the same name in PATH. (#543)
```

Comment:
```suggestion
* Hardcode paths to Apple tools (`productbuild`, `productsign`, `codesign`) to avoid using other tools with the same name in PATH. (#543)
```

Reply:
done